doc/development/06_ca_manager.md

```diff
@@ -0,0 +1,978 @@
+
+The following functions are used to manage parents of CAs.
+
+```rust
+/// # CAs as children
+///
+impl CaManager {
+    /// Adds a parent to a CA. This will trigger that the CA connects to this new parent
+    /// in order to learn its resource entitlements and set up the resource class(es) under
+    /// this parent, and request certificate(s).
+    pub async fn ca_parent_add(
+        &self,
+        handle: Handle,
+        parent: ParentCaReq,
+        actor: &Actor
+    )-> KrillResult<()> { ... }
+
+    /// Removes a parent from a CA, this will trigger that best effort revocations of existing
+    /// keys under this parent are requested. Any resource classes under the parent will be removed
+    /// and all relevant content will be withdrawn from the repository.
+    pub async fn ca_parent_remove(
+        &self,
+        handle: Handle,
+        parent: ParentHandle,
+        actor: &Actor
+    ) -> KrillResult<()> { ... }
+
+    /// Updates a parent of a CA, this can be used to update the service uri and/or
+    /// identity certificate for an existing parent.
+    pub async fn ca_parent_update(
+        &self,
+        handle: Handle,
+        parent: ParentHandle,
+        contact: ParentCaContact,
+        actor: &Actor,
+    ) -> KrillResult<()> { ... }
+
+    /// Returns the parent statuses for this CA
+    pub async fn ca_parent_statuses(&self, ca: &Handle) -> KrillResult<ParentStatuses> { ... }
+
+    /// Refresh all CAs:
+    /// - process all CAs in parallel
+    /// - process all parents for CAs in parallel
+    ///    - send pending requests if present, or
+    ///    - ask parent for updates and process if present
+    pub async fn cas_refresh_all(&self, actor: &Actor) { ... }
+
+    /// Synchronizes a CA with one of its parents:
+    ///   - send pending requests if present; otherwise
+    ///   - get and process updated entitlements
+    ///
+    /// Note: if new request events are generated as a result of processing updated entitlements
+    ///       then they will trigger that this synchronization is called again so that the pending
+    ///       requests can be sent.
+    pub async fn ca_sync_parent(&self, handle: &Handle, parent: &ParentHandle, actor: &Actor) -> KrillResult<()> { ... }
+}
+```
+
+
+CA as Parent Related Functions
+------------------------------
+
+The following functions are used to manage children of CAs.
+
+```rust
+/// # CAs as children
+///
+impl CaManager {
+    /// Adds a child under a CA. The 'service_uri' is used here so that
+    /// the appropriate `ParentCaContact` can be returned. If the `AddChildRequest`
+    /// contains resources not held by this CA, then an `Error::CaChildExtraResources`
+    /// is returned.
+    pub async fn ca_add_child(
+        &self,
+        ca: &Handle,
+        req: AddChildRequest,
+        service_uri: &uri::Https,
+        actor: &Actor,
+    ) -> KrillResult<ParentCaContact> { ... }
+
+    /// Show details for a child under the TA.
```

Comment:
Why under the TA? Doesn't it report details for the child under the given CA handle?